src/wallet/rpcdump.cpp

```diff
@@ -614,8 +631,9 @@ UniValue dumpwallet_impl(const UniValue& params, bool fHelp, bool fDumpZKeys)
             libzcash::SaplingExtendedSpendingKey extsk;
             if (pwalletMain->GetSaplingExtendedSpendingKey(addr, extsk)) {
                 auto ivk = extsk.expsk.full_viewing_key().in_viewing_key();
-                std::string strTime = EncodeDumpTime(pwalletMain->mapSaplingZKeyMetadata[ivk].nCreateTime);
-                file << strprintf("%s %s # zaddr=%s\n", EncodeSpendingKey(extsk), strTime, EncodePaymentAddress(addr));
+                CKeyMetadata keyMeta = pwalletMain->mapSaplingZKeyMetadata[ivk];
+                std::string strTime = EncodeDumpTime(keyMeta.nCreateTime);
+                file << strprintf("%s %s %s %s # zaddr=%s\n", EncodeSpendingKey(extsk), strTime, keyMeta.hdKeypath, keyMeta.seedFp.GetHex(), EncodePaymentAddress(addr));
```

Comment:
What will this do if `hdKeypath` or `seedFp` are empty/null (as for a key imported via `z_importkey`)? If empty I assume we just get additional whitespace in the file, but `seedFp.IsNull()` is when it is the all-zeroes array, which I think would get printed out here. We might need to be more verbose here in order to only add these two if they are meaningfully-filled.

Reply:
In fact, I think the import logic will break with this as-is. `hdKeypath` will be empty, and thus nothing prints for it, but `seedFp` will print all-zeroes hex. Then on importing, `seedFp` will be parsed as the third item in the list, and thus be incorrectly interpreted as `hdKeypath`.